### PR TITLE
Loki Monaco Editor: grab operator documentation from the operations module

### LINKED ISSUE #57396: Loki: Add tests for public/app/plugins/datasource/loki/querybuilder/operations.ts
Currently we don't have unit tests for `operations.ts`. We should make sure this module has enough coverage, by asserting that `getOperationDefinitions()` returns a list of `QueryBuilderOperationDef[]`.

### PR DESCRIPTION
**Pull request by matyax**
**What this PR does / why we need it**:

This PR address the following comment https://github.com/grafana/grafana/pull/57186#pullrequestreview-1147201918 by implementing a method to grab the explain from the opeartions module, thus keeping the definitions in a single place.

**Which issue(s) this PR fixes**:

Part of https://github.com/grafana/grafana/issues/44985
Fixes https://github.com/grafana/grafana/issues/57396

**Special notes for your reviewer**:

https://user-images.githubusercontent.com/1069378/197527311-edc73230-cd18-4742-8678-9aee7fcdd13c.mov

### CODE REVIEW

File: public/app/plugins/datasource/loki/querybuilder/operations.test.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  const definitions = getOperationDefinitions();
+  for (const definition of definitions) {
+    if (!undocumentedOperationsIds.includes(definition.id)) {
+      operations.push(definition.id);
+    } else {
+      undocumentedOperations.push(definition.id);
+    }
+  }
+
+  test('Resolvs operation definitions', () => {
```
Comment (ivanahuckova) [inline public/app/plugins/datasource/loki/querybuilder/operations.test.ts:33]:
```suggestion
  test('Resolves operation definitions', () => {
```

File: public/app/plugins/datasource/loki/querybuilder/operations.ts [hunk 2]
```diff
@@ -420,3 +420,19 @@ export function getOperationDefinitions(): QueryBuilderOperationDef[] {
...
 }
+
+// Keeping a local copy as an optimization measure.
+const definitions = getOperationDefinitions();
+
+/**
+ * Given an operator, return the corresponding explain.
+ * Conceibed for usage within the Query Editor.
+ */
+export function explainOperator(id: LokiOperationId | string): string {
```
Comment (ivanahuckova) [inline public/app/plugins/datasource/loki/querybuilder/operations.ts:431]: I am wondering if this should be `explainOperation` or `explainCompletionItem` as it explains different operations and not just operators.

File: public/app/plugins/datasource/loki/querybuilder/operations.ts [hunk 3]
```diff
@@ -420,3 +420,19 @@ export function getOperationDefinitions(): QueryBuilderOperationDef[] {
 
   return list;
 }
+
+// Keeping a local copy as an optimization measure.
+const definitions = getOperationDefinitions();
+
+/**
+ * Given an operator, return the corresponding explain.
+ * Conceibed for usage within the Query Editor.
```
Comment (ivanahuckova) [inline public/app/plugins/datasource/loki/querybuilder/operations.ts:429]:
```suggestion
 * For usage within the Query Editor.
```

**Review by ivanahuckova — APPROVED** [review]
LGTM, left some feedback bellow. I am wondering, if there is a plan to also use documentation from `getOperationDefinitions` for functions?

File: public/app/plugins/datasource/loki/querybuilder/operations.ts (hunk 2 above)
Comment (matyax) [reply public/app/plugins/datasource/loki/querybuilder/operations.ts:431]: The whole thing is called `operations`, so I followed that pattern.

**Comment by matyax** [comment]
> I am wondering, if there is a plan to also use documentation from getOperationDefinitions for functions?

@ivanahuckova I already have access for the documentation from `syntax.ts`.

imagen

I'm not sure what's the difference and which one would be preferred. I would go with that you would suggest.

**Comment by matyax** [comment]
I'm going to merge this one, let me know if you want to change the docs for the functions. Thanks!

File: public/app/plugins/datasource/loki/querybuilder/operations.ts (hunk 2 above)
Comment (ivanahuckova) [reply public/app/plugins/datasource/loki/querybuilder/operations.ts:431]: Not sure I understand what you mean here.  In my opinion, `explainOperator` should be `explainOperation` as it explains any operations, not just operators such as `=`, `!=`,....

Comment (matyax) [reply public/app/plugins/datasource/loki/querybuilder/operations.ts:431]: Ohhhh, gotcha, I misread your comment. `explainOperation` is a good suggestion. I'll update it in one of my open PRs. Thanks!

**Pull request merged**